test/unit/org/apache/cassandra/inject/Injections.java

```diff
@@ -0,0 +1,598 @@
+        {
+            internalBarrier.await(false, true);
+        }
+
+        /**
+         * Just decrement the number of parties this barrier is waiting for but do not wait regardless of how
+         * this barrier was setup - this is force countdown.
+         */
+        public void countDown()
+        {
+            try
+            {
+                internalBarrier.await(true, false);
+            }
+            catch (InterruptedException ignored)
+            {
+                // will not happen in case just a countdown
+            }
+        }
+
+        /**
+         * Get the number of parties the barrier is currently awaiting for.
+         */
+        public long getCount()
+        {
+            return internalBarrier.getCount();
+        }
+
+        /**
+         * Reset the barrier, that is, let all waiting threads resume and reset the number of parties
+         * (if cyclic is enabled).
+         */
+        public void reset()
+        {
+            internalBarrier.reset();
+        }
+
+        @CallMe
+        public static void doAction(String name, boolean doCountDown, boolean doAwait) throws InterruptedException
+        {
+            CyclicBarrier barrier = barriers.get(name);
+            barrier.await(doCountDown, doAwait);
+        }
+
+        public static class BarrierBuilder extends CrossProductInjectionBuilder<Barrier, BarrierBuilder>
+        {
+            private final String name;
+            private final int parties;
+            private final boolean cyclic;
+            private final boolean doAwait;
+            private final boolean doCountDown;
+
+            private BarrierBuilder(String name, int parties, boolean cyclic, boolean doCountDown, boolean doAwait)
+            {
+                super(String.format("barrier/%s/%s", name, UUID.randomUUID().toString()));
+                this.name = name;
+                this.parties = parties;
+                this.cyclic = cyclic;
+                this.doAwait = doAwait;
+                this.doCountDown = doCountDown;
+                add(newActionBuilder().actions().doAction(method(Barrier.class, CallMe.class)
+                        .args(quote(name), doCountDown, doAwait)));
+            }
+
+            @Override
+            public Barrier build()
+            {
+                return new Barrier(id, name, parties, cyclic, doCountDown, doAwait, getRules());
+            }
+        }
+    }
+
+    /**
+     * Creates {@link Times} injection.
+     *
+     * @param name name of the internal counter
+     * @param defaultTimes the number of times the action should be executed
+     */
+    public static Times.TimesBuilder newTimes(String name, int defaultTimes)
+    {
+        return new Times.TimesBuilder(name, defaultTimes);
+    }
+
+    /**
+     * Creates an injection which allows to invoke a defined action for a defined number of times.
+     */
+    public static class Times extends Injection
```

Comment:
nit: Ignore me if `Times` is used in a later SAI-related patch, but if it isn't used even then, might as well remove the dead code. There are a couple other things in here (ex. `PauseBuilder` being unused)

Reply:
`Times` definitely, it isn't used. `Pause` is used later on for introducing delays in index writing tests.